quinn-proto/src/connection/streams/state.rs

```diff
@@ -443,32 +443,27 @@ impl StreamsState {
             stats.max_stream_data += 1;
         }
 
-        // MAX_STREAMS_UNI
-        if pending.max_uni_stream_id && buf.len() + 9 < max_size {
-            pending.max_uni_stream_id = false;
-            retransmits.get_or_create().max_uni_stream_id = true;
-            self.max_streams_dirty[Dir::Uni as usize] = false;
-            trace!(
-                value = self.max_remote[Dir::Uni as usize],
-                "MAX_STREAMS (unidirectional)"
-            );
-            buf.write(frame::Type::MAX_STREAMS_UNI);
-            buf.write_var(self.max_remote[Dir::Uni as usize]);
-            stats.max_streams_uni += 1;
-        }
-
-        // MAX_STREAMS_BIDI
-        if pending.max_bi_stream_id && buf.len() + 9 < max_size {
-            pending.max_bi_stream_id = false;
-            retransmits.get_or_create().max_bi_stream_id = true;
-            self.max_streams_dirty[Dir::Bi as usize] = false;
-            trace!(
-                value = self.max_remote[Dir::Bi as usize],
-                "MAX_STREAMS (bidirectional)"
-            );
-            buf.write(frame::Type::MAX_STREAMS_BIDI);
-            buf.write_var(self.max_remote[Dir::Bi as usize]);
-            stats.max_streams_bidi += 1;
+        for dir in Dir::iter() {
+            // MAX_STREAMS
+            if pending.max_stream_id[dir as usize] && buf.len() + 9 < max_size {
```

Comment:
Style: this looks like an opportunity for early continue?

Reply:
Good catch, fixed.